### ISSUE TITLE
Enhance sidebar nav to take another level of hierarchy

### ISSUE THREAD

**Issue by crandmck** [loopbackio/loopback.io#66]
Currently, the sidebar nav only allows three levels of hierarchy:
- Top-level "folders"
  - Second-level "subfolders"
    - Third-level "subfolder items"

This works for most of our content, but there are a few areas where we could use one more level.  The code that processes the `yml` sidebar file does not currently support this.  

In addition, there is an oddity left over from the original implementation where the _first_ item in a "folder" cannot be a "subfolder" (only a non-container leaf node is allowed).  For example, I had to stick a dummy item http://loopback.io/doc/en/lb2/Defining-models.html under the container, because the first item, [Creating models](http://loopback.io/doc/en/lb2/Creating-models.html), is a "subfolder". Without the intervening item, jekyll gives a build error:

```
           Error: (/Users/rand/StrongLoop/loopback.io/_data/sidebars/lb2_sidebar.yml): did not find expected key while parsing a block mapping at line 236 column 5
```

I would like to be able to make an article be a "container" in the sidebar, regardless of where it occurs.

**Comment by Sequoia** [comment]
[`links.html`](https://github.com/strongloop/loopback.io/blob/gh-pages/_includes/links.html#L3-L20) purports to also use the sidebar menu scheme (which I'm revamping), and it looks like [all the pages in `/pages/en/contrib/`](https://github.com/strongloop/loopback.io/blob/gh-pages/pages/en/contrib/) _include_ the file, but **it appears to do nothing**.

To wit, when I add `<hr />`s at the top and bottom of `links.html`, the output looks like this:

![](https://i.imgur.com/r8r3oqK.png)

Note that there is nothing there.

For this reason I **do not** plan to update `links.html` at this time, as it's not clear that it actually does anything. Please let me know if you know what it does & it's important to update (this can always be done after).

**Comment by Sequoia** [comment]
Other templates that use the existing menu YAML schema:
- [`tocpage.html`](https://github.com/strongloop/loopback.io/blob/gh-pages/scripts-pdf/pdfconfigs/tocpage.html): Looks like its for PDF generation. Will update
- [`prince-list.txt`](https://github.com/strongloop/loopback.io/blob/gh-pages/scripts-pdf/pdfconfigs/prince-list.txt): Also appears to be related to PDF generation? Will update as well (or investigate)
- [`topnav.html`](https://github.com/strongloop/loopback.io/blob/gh-pages/_includes/topnav.html): Only reads [`topnav.yaml`](https://github.com/strongloop/loopback.io/blob/gh-pages/_data/topnav.yml), which I do not plan to update. I **do not plan** to update this file because it's unrelated to and separate from other menus.

**Comment by crandmck** [comment]
Sounds good.  BTW, `prince-list.txt` is explained in http://idratherbewriting.com/documentation-theme-jekyll/mydoc_generating_pdfs.html.

**Comment by Sequoia** [comment]
See #73

**Comment by crandmck** [comment]
This fixed in #73.
